nodejs/eks/workerpool.ts

```diff
@@ -49,11 +44,26 @@ export interface WorkerPoolOptions {
      */
     cluster: aws.eks.Cluster;
 
+    /**
+     * k8s provider for this worker pool
+     */
+    provider: k8s.Provider;
```

Comment:
`provider` should not be a part of `WorkerPoolOptions`: it is already part of `pulumi.ComponentResourceOptions`, which is passed to the `WorkerPool` constructor. I'd rather we leave it as a separate parameter on `createWorkerPool`.

Reply:
I see, I did not know that. What's the right way to get the k8s provider from within the constructor through ComponentResourceOptions? The original `this.getProvider("kubernetes:core:v1/ConfigMap")` code results in type error.

Reply:
What was the type error? That method should exist.

Reply:
If the type error was complaining about the value possibly being `undefined`, you can do something like:

```
const k8sProvider = this.getProvider("kubernetes:core:v1/ConfigMap");
if (k8sProvider === undefined) {
    throw new RunError("a 'kubernetes' provider must be specified for a 'WorkerPool'");
}
```

Reply:
This is exactly what I needed, thanks a lot for the tip :)